cirq/devices/grid_qubit.py

```diff
@@ -38,6 +38,93 @@ def is_adjacent(self, other: ops.Qid) -> bool:
         return (isinstance(other, GridQubit) and
                 abs(self.row - other.row) + abs(self.col - other.col) == 1)
 
+    @staticmethod
+    def square(diameter: int, top: int = 0, left: int = 0) -> List['GridQubit']:
+        """Returns a square of GridQubits.
+
+        Args:
+            diameter: Length of a side of the square
+            top: Row number of the topmost row
+            left: Column number of the leftmost row
+
+        Returns:
+            A list of GridQubits filling in a square grid
+        """
+        return GridQubit.rect(diameter, diameter, top=top, left=left)
+
+    @staticmethod
+    def rect(rows: int, cols: int, top: int = 0,
+             left: int = 0) -> List['GridQubit']:
+        """Returns a rectangle of GridQubits.
+
+        Args:
+            rows: Number of rows in the rectangle
+            cols: Number of columns in the rectangle
+            top: Row number of the topmost row
+            left: Column number of the leftmost row
+
+        Returns:
+            A list of GridQubits filling in a rectangular grid
```

Comment:
I wonder if it would be more useful if this were a matrix and not a list.  That way I could do q[0][1] to get row 0 and column 1?

Reply:
That sounds reasonable. I'll change it to a list of lists.

Reply:
Most methods that we have take a list instead of a list of lists of qubits. If you want random access into the grid, it seems like you would just use GridQubit directly anyway?

I vote leave it as a list, because then the signature matches from_diagram.

Reply:
We now have test passing commits either way. Whenever it is decided, we can just pick the corresponding commit.

Reply:
qubits = cirq.GridQubit.square(5)
then accessing qubits[i][j] is slightly nicer than cirq.GriqQubit(i,j).
but I think actually there is a bunch of stuff that could naturally go in an object that you produce via GridQubit... i.e. a grid.  Now that I've written a could versions of variational algorithms I find myself doing a lot of 
```
for x in range(d):
  for y in range(d-1):
      nearest neighbor
```
This gets even worse with the odder sized lattice.

So for now probably list is good, but I think we may need an actual Grid object helper.  I'll open an issue for that.

Reply:
Let's go with @Strilanc 's suggestion.  Sorry for the detour.

Reply:
I've reverted the commits to the List[GridQubit] return type version.